storage/client/src/main/java/com/azure/storage/blob/BlobAsyncClient.java

```diff
@@ -51,15 +66,19 @@
  * object through {@link Mono#toFuture()}.
  */
 public class BlobAsyncClient {
+    private static final long BLOB_DEFAULT_DOWNLOAD_BLOCK_SIZE = 4 * Constants.MB;
 
-    protected BlobAsyncRawClient blobAsyncRawClient;
+    final BlobAsyncRawClient blobAsyncRawClient;
+
+    private final String snapshot;
 
     /**
      * Package-private constructor for use by {@link BlobClientBuilder}.
-     * @param azureBlobStorage the API client for blob storage API
+     * @param azureBlobStorageBuilder the API client builder for blob storage API
      */
-    BlobAsyncClient(AzureBlobStorageImpl azureBlobStorage) {
-        blobAsyncRawClient = new BlobAsyncRawClient(azureBlobStorage);
+    BlobAsyncClient(AzureBlobStorageBuilder azureBlobStorageBuilder, String snapshot) {
+        this.blobAsyncRawClient = new BlobAsyncRawClient(azureBlobStorageBuilder.build());
+        this.snapshot = snapshot;
```

Comment:
Talked to James in person - this behavior is having the equivalent effect of appending the snapshot on the URL. And since we are already parsing the URL to get the container name and blob name, this does not introduce a performance overhead.
Also, this is also actually preferred, compare to the previous way to doing it, in the way that having a query parameter decoupled from the base URL often help prevent potential URL parsing issues, when another query parameter needs to be appended.